### ISSUE TITLE
[1.48] Unobtainable coins

### ISSUE THREAD

**Issue by MithranArkanere** [ochadenas/cpudefense#189]
Coins become unobtainable in endless after choosing replay stage, or leaving to main menu and replaying the level.

The "x ox x coins got" message disappears altogether. 

This does not happen when replaying the level after defeat.
Losing on purpose after the coins are bugged does not restore the count.

**Comment by ochadenas** [comment]
Could you confirm that this is really version 1.48 where you encountered this bug?

Because this bug actually existed, but I thought that had been fixed in 1.48, and I can't reproduce it in the current version. Could you explain exactly what you have done in which order?

**Comment by MithranArkanere** [comment]
[Image]

1.48 for sure.
To replicate this bug, play any Endless level that has never been played before, and before having finished that level once, leave the level or choose to replay the level. When you return to that level, the coins will be gone.
________________________________

**Comment by MithranArkanere** [comment]
![Screenshot_20241204_215225_ChipDefense](https://github.com/user-attachments/assets/e0120273-3351-48a7-b3b4-f5949c40261d)
The image didn't upload via mail response.

**Comment by MithranArkanere** [comment]
This is fixed in versikn 1.49.

**Comment by MithranArkanere** [comment]
False... what's the opposite of alarm? 

This is still hapoening in 1.50.

![Screenshot_20241220_204922_ChipDefense](https://github.com/user-attachments/assets/8016a10e-c2c4-42a0-994f-761411e9b3a3)

**Comment by ochadenas** [comment]
You are right. I was able to reproduce the effect. Will have a look into this.

**Comment by carlosjeurissen** [comment]
@ochadenas wonderful! Just to be sure you also noticed, in addition to the empty coins message, the message for 1 out of 3 coins has double the word "got" in it. Which seems like a bug.

**Comment by ochadenas** [comment]
Yes, I've seen that in your screenshot. Already fixed.

**Comment by ochadenas** [comment]
I think I've found the bug. The number of available coins is only  saved on level completion, not on entering the level. And when you leave the stage to the main menu and then choose "resume game", the game no longer knows about the coins. 

"Replay stage" works, however. In that case the coins are calculated once again.

**Comment by ochadenas** [comment]
Should be fixed in 1.51. Please re-open if the problem still persists.
